airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py

```diff
@@ -249,25 +279,139 @@ async def test_state_with_abnormally_large_values(
+        # with the current batch's stream state and then use it in the following read() request
+        current_state = state_message.state
+        if current_state and current_state.type == AirbyteStateType.STREAM:
+            per_stream = current_state.stream
+            if per_stream.stream_state:
+                stream_name_to_per_stream_state[per_stream.stream_descriptor.name] = (
+                    per_stream.stream_state.dict() if per_stream.stream_state else {}
+                )
+        state_input = [
+            {"type": "STREAM", "stream": {"stream_descriptor": {"name": stream_name}, "stream_state": stream_state}}
+            for stream_name, stream_state in stream_name_to_per_stream_state.items()
+        ]
+        return state_input, stream_name_to_per_stream_state
+
+    def get_unique_state_messages(self, states: List[AirbyteStateMessage]) -> List[AirbyteStateMessage]:
+        """
+        Validates a list of state messages to ensure that consecutive messages with the same stream state are represented by only the first message, while subsequent duplicates are ignored.
+        """
+        # If there is only one state message or less, the list is considered to be unique
+        if len(states) <= 1:
+            return states
+
+        stream_state_attr = "state.stream.stream_state"
+        record_count_attr = "state.sourceStats.recordCount"
+
+        # Define a function to get the stream state attribute from an AirbyteStateMessage object
+        get_state = operator.attrgetter(stream_state_attr)
+
+        # Define a function to get the record count attribute from an AirbyteStateMessage object
+        get_record_count = operator.attrgetter(record_count_attr)
+
+        current_idx = 0
+        result = []
+
+        # Iterate through the list of state messages
+        while current_idx < len(states) - 1:
+            next_idx = current_idx + 1
+            # Check if consecutive messages have the same stream state
+            while get_state(states[current_idx]) == get_state(states[next_idx]) and next_idx < len(states) - 1:
+                next_idx += 1
+
+            states[current_idx].state.sourceStats = AirbyteStateStats(recordCount=sum(map(get_record_count, states[current_idx:next_idx])))
+            # Append the first message with a unique stream state to the result list
+            result.append(states[current_idx])
+            # If the last message has a different stream state than the previous one, append it to the result list
+            if next_idx == len(states) - 1 and get_state(states[current_idx]) != get_state(states[next_idx]):
+                result.append(states[next_idx])
+            current_idx = next_idx
+
+        return list(self.get_expected_record_count_per_state(result))
+
+    def get_expected_record_count_per_state(self, states: List[AirbyteStateMessage]) -> Iterator[Tuple[AirbyteStateMessage, float]]:
```

Comment:
not: shouldn't the second value of the tuple be an integer since record counts can only be natural numbers?

Reply:
I thought the same way but decided to use the same type as provided by this model:
```
class AirbyteStateStats(BaseModel):
    class Config:
        extra = Extra.allow

    recordCount: Optional[float] = Field(
        None,
        description='the number of records which were emitted for this state message, for this stream or global',
    )
```

Reply:
this makes sense. thank you!